src/main/java/com/aws/greengrass/mqtt/bridge/auth/MQTTClientKeyStore.java

```diff
@@ -128,51 +128,52 @@ public void updateCA(List<String> caCerts) throws IOException, CertificateExcept
                 keyStore.deleteEntry(alias);
             }
         }
-
         for (int i = 0; i < caCerts.size(); i++) {
             X509Certificate caCert = pemToX509Certificate(caCerts.get(i));
             keyStore.setCertificateEntry("CA" + i, caCert);
         }
-
+        setCaCerts(caCerts);
         updateListeners.forEach(UpdateListener::onCAUpdate); //notify MQTTClient
     }
 
+    private void setCaCerts(List<String> caCerts) {
+        synchronized (this.caCerts) {
+            this.caCerts.clear();
+            this.caCerts.addAll(caCerts);
+        }
+    }
+
+    private List<String> getCaCerts() {
+        synchronized (caCerts) {
+            return new ArrayList<>(caCerts);
+        }
+    }
+
     /**
-     * Get the CA cert chain from the key store.
+     * Return CA cert chain as a string.
      *
-     * @return ca cert chain
-     * @throws KeyStoreException if keystore has not been initialized
-     * @throws CertificateEncodingException if an encoding error occurs
+     * @return ca certs
      */
-    @SuppressWarnings("PMD.InsufficientStringBufferDeclaration")
-    public Optional<String> getCACertChain() throws KeyStoreException, CertificateEncodingException {
-        List<Certificate> caCerts = getCACerts();
-        if (caCerts.isEmpty()) {
+    public Optional<String> getCaCertsAsString() {
+        List<String> certs = getCaCerts();
+        if (certs.isEmpty()) {
             return Optional.empty();
         }
-        StringBuilder chain = new StringBuilder("-----BEGIN CERTIFICATE-----");
-        for (Certificate cert : caCerts) {
-            String encodedCert = Base64.getEncoder().encodeToString(cert.getEncoded());
-            chain.append(System.lineSeparator())
-                    .append(encodedCert);
+        StringBuilder result = new StringBuilder(BEGIN_CERT);
+        for (String cert : certs) {
```

Comment:
this doesn't look valid. For a chain, it should be multiple begin and end certs

Reply:
fixed, looking at an effectiveConfig.yaml from a previous run, looks like each cert ends with a newline, so opted to just append the certs together